Cakefile

```diff
@@ -400,7 +400,8 @@ task 'bench', 'quick benchmark of compilation time', ->
 
 
 # Run the CoffeeScript test suite.
-runTests = (CoffeeScript) ->
+runTests = (CoffeeScript, pattern="") ->
+  pattern = new RegExp pattern
```

Comment:
This feels like a separate PR. Also, why is it needed? For me on `main`, `cake test` takes about 6 seconds. Are the new tests in this PR slowing the tests significantly, with all the child processes being created?

Reply:
I'm happy to split this out. The main issue isn't the speed difference but the ability to isolate the output for a few tests when debugging. When the stack traces changed it caused almost all the error message tests to fail. When adding detailed `console.log` debugging sometimes the output would scroll beyond the capacity of the terminal. Being able to run a single test or subset of tests can make it easier to see the specific error.

Any thoughts on using Mocha as a test runner? (As a separate PR of cources)

Reply:
> Any thoughts on using Mocha as a test runner?

Part of the ethos of this project has been that it has no dependencies (at least, other than dev dependencies) and that the CoffeeScript repo itself should serve as an example of how to build a robust application using CoffeeScript. Along those lines, I think there would be many who would object to adding a large dependency like that for something as fundamental as running tests; notice how we don’t have a library for generating the static website, either.

I could be wrong, though, if the benefits were significant. Personally the refactoring required to add Mocha feels like time that could be better spent on other enhancements.